atomic_reactor/util.py

```diff
@@ -808,11 +809,59 @@ def query_registry(registry_session, image, digest=None, version='v1', is_blob=F
     logger.debug("query_registry: querying {}, headers: {}".format(url, headers))
 
     response = registry_session.get(url, headers=headers)
+    for r in chain(reversed(response.history), [response]):
+        logger.debug("query_registry: [%s] %s", r.status_code, r.url)
+
+    logger.debug("query_registry: response headers: %s", response.headers)
     response.raise_for_status()
 
     return response
 
 
+def guess_manifest_media_type(content):
+    encoding = guess_json_utf(content)
+    try:
+        manifest = json.loads(content.decode(encoding))
+    except (ValueError,           # Not valid JSON
+            TypeError,            # Not an object
+            UnicodeDecodeError):  # Unable to decode the bytes
+        logger.exception("Unable to decode JSON")
+        logger.debug("response content (%s): %r", encoding, content)
+        return None
+
+    try:
+        return manifest['mediaType']
+    except KeyError:
+        # no mediaType key
+        if 'schemaVersion' in manifest and manifest['schemaVersion'] == 1:
+            return get_manifest_media_type('v1')
+
+        logger.warning("no mediaType or schemaVersion=1 in manifest, keys: %s",
+                       manifest.keys())
+
+
+def manifest_is_media_type(response, media_type):
+    try:
+        received_media_type = response.headers['Content-Type']
+    except KeyError:
+        # Guess media type from content
+        logger.debug("No Content-Type header; inspecting content")
+        received_media_type = guess_manifest_media_type(response.content)
+        logger.debug("guessed media type: %s", received_media_type)
+
+    if received_media_type is None:
```

Comment:
What if `media_type` is None? This method will incorrectly return `False`. Not sure if it's worth handling that.